scripts/live-canary/notify_slack.py

```
@@ -818,6 +835,203 @@ def slack_payload(
+    *,
+    repo: str,
+    github_token: str,
+    run_context: CanaryRunContext,
+    run_url: str | None,
+    commit: str | None,
+    category_summary: str = "",
+) -> str:
+    if not run_context.target_pr:
+        return ""
+    if not run_context.target_pr.isdecimal():
+        raise ValueError("target_pr must be a decimal pull request number")
```

Ruff (0.15.20)

[warning] Avoid specifying long messages outside the exception class

(TRY003)